src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreDatabaseTaskDataObject.cs

```diff
@@ -195,8 +195,12 @@ public override Server Server
         /// Add a backup file to restore plan media list
         /// </summary>
         /// <param name="filePaths"></param>
-        public void AddFiles(string filePaths)
+        public void AddDevices(string filePaths, DeviceType deviceType)
         {
+            if (deviceType != DeviceType.File && deviceType != DeviceType.Url)
```

Comment:
Please add a comment about why you're doing this. It seems like we should be throwing an exception if we get a device type we don't support instead of forcing it to be a file though...